### PR TITLE
feat: wakuv2 peer persistence

### PR DESCRIPTION
**Pull request by richard-ramos**

### CODE REVIEW

File: wakuv2/waku.go [hunk 1]
```diff
@@ -165,15 +170,35 @@ func New(nodeKey string, cfg *Config, logger *zap.Logger) (*Waku, error) {
...
 	connStatusChan := make(chan node.ConnStatus, 100)
 
 	if cfg.KeepAliveInterval == 0 {
 		cfg.KeepAliveInterval = DefaultConfig.KeepAliveInterval
 	}
 
+	libp2pOpts := node.DefaultLibP2POptions
+	libp2pOpts = append(libp2pOpts, libp2p.BandwidthReporter(waku.bandwidthCounter))
+
+	if appdb != nil && cfg.PersistPeers {
```
Comment (alaibe) [inline wakuv2/waku.go]: I think we could do:
```
if cfg.PersistPeers {
  if appdb == nil {
    return nil, fmt.Errorf(¨A db connection must be provided in order to persist the peers") // or a better message
  }
  ...rest of logic
```

That way we will always respect the config and not have a state where we want to persist the peers and not doing it

File: wakuv2/persistence/sqlcipher.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
+package persistence
```
Comment (alaibe) [inline wakuv2/persistence/sqlcipher.go]: nit: why is this file named sqlcipher and not query.go?
I don't see any sqlcipher related items here?

**Review by alaibe — APPROVED** [review]
LGTM, but pls see the 2 comments below

File: wakuv2/persistence/sqlcipher.go (hunk 2 above)
Comment (richard-ramos) [reply wakuv2/persistence/sqlcipher.go]: originally it had, but I removed the code but forgot to rename the file! :)

**Pull request merged**